README.rst

```diff
@@ -827,6 +828,89 @@ The filter can have the following parameters:
   Logstash section.
 
 
+.. _encode-lua:
+
+encode_lua
+^^^^^^^^^^
+
+This filter helps to create configuration in a Lua friendly format.
+The expected data structure is the following:
+
+.. code:: yaml
+
+    my_lua_config:
+      fork: false
+      external_addresses:
+        - 1.2.3.4
+        - 5.6.7.8
+      admins:
+        - admin@example.com
+      contact_info:
+        abuse: abuse@example.com
+        admin: admin@example.com
+
+Lua is a small scripting language, often embedded into C/C++ applications.
+This encoder does a best effort to match configuration files seen in the wild,
+while allowing the user to further customize how the final output is rendered.
+
+The above variable can be used in the template file like this:
+
+.. code:: jinja2
+
+    {{ my_lua_config | encode_lua }}
+
+The output of such a template would be:
+
+.. code:: lua
+
+    fork = false;
+    external_addresses = {
+        "1.2.3.4";
+        "5.6.7.8";
+    }
+    admins = {
+        "admin@example.com";
+    }
+    contact_info = {
+        abuse = "abuse@example.com";
+        admin = "admin@example.com";
+    }
+
+The filter can have the following parameters:
+
+- ``convert_bools=false``
+
+  Indicates whether Boolean values presented as a string should be
+  converted to a real Boolean value. For example ``var1: 'True'`` would
+  be represented as a string but by using the ``convert_bools=true`` it
+  will be converted into Boolean like it would be defined like ``var1:
+  true``.
+
+- ``convert_nums=false``
+
+  Indicates whether number presented as a string should be converted to
+  number. For example ``var1: '123'`` would be represented as a string
+  but by using the ``convert_nums=true`` it will be converted it to a
+  number like it would be defined like ``var1: 123``. It's also possible
+  to use the YAML type casting to convert string to number (e.g. ``!!int
+  "1234"``, ``!!float "3.14"``).
+
+- ``sort_keys=false``
+
+  Indicates whether the keys should be sorted when the output is rendered,
+  or left to python's implicit handling of dict ordering.
```

Comment:
I believe this should be the default and the only possible formatting as dictionaries don't preserve order in Python which might result in creation of non-immutable configuration files.

Reply:
Since 3.7, [python has gone so far as to codify that dict order will be preserved](https://docs.python.org/3/whatsnew/3.7.html), as was 99.9% the case in previous versions.

> the insertion-order preservation nature of [dict](https://docs.python.org/3/library/stdtypes.html#typesmapping) objects [has been declared](https://mail.python.org/pipermail/python-dev/2017-December/151283.html) to be an official part of the Python language spec.

Would you be okay with setting it to `True` but leaving the option?

On a personal note, I think you should consider adding this sort option to your other filters.

Reply:
@jtyr I've pushed a change with it defaulting to `True`. Let me know what you think. Thanks again :)